### ISSUE TITLE
EXCEPTION: NSt8ios_base7failureB5cxx11E

### ISSUE THREAD

**Issue by sammy007** [zcash/zcash#2479]
```
EXCEPTION: NSt8ios_base7failureB5cxx11E String length limit exceeded: iostream error Zcash in ProcessMessages()
```

v1.0.10-1

**Comment by daira** [comment]
In Zcash 1.0.10 there was an unintended and incorrect change to the encoding of a "version" message. This was corrected in Zcash 1.0.10-1 back to the encoding used by versions up to and including 1.0.9.

However, when a non-1.0.10 node (including 1.0.10-1) receives a connection from a 1.0.10 node, it will log this exception (or another exception depending on the precise random bytes used in the ``nonce`` field of the "version" message) due to the incorrect encoding used by the latter. The non-1.0.10 node is behaving correctly in this case.

These messages will go away once remaining 1.0.10 nodes are upgraded. We might change the error reporting for 1.0.11 in order to avoid causing any unnecessary alarm, but this condition is basically harmless.

**Comment by str4d** [comment]
When we send our version message first, and it can't be parsed by the 1.0.10 peer, it will send back a reject message, and we treat that as misbehaving because it hasn't sent a version message. Upstream recently fixed this in https://github.com/bitcoin/bitcoin/pull/9720, which at a minimum requires the following PRs to be merged first:

- https://github.com/bitcoin/bitcoin/pull/7181
- https://github.com/bitcoin/bitcoin/pull/9260

**Comment by nathan-at-least** [comment]
Is #2491 a proposed fix for this issue?

**Comment by daira** [comment]
No, #2491 did not cover this case. There is still a bug to be fixed here (independent of it being triggered by 1.0.10).

**Comment by daira** [comment]
1.0.10 [has auto-senesced](https://github.com/zcash/zcash/pull/2491#issuecomment-331429949) so there is no point in fixing just the specific case that 1.0.10 triggers, but the response is still incorrect as pointed out by @str4d, i.e. we should still fix https://github.com/bitcoin/bitcoin/pull/9720 .
